src/arrayMethodSort.js

```diff
@@ -4,8 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = 0; j < length - 1 - i; j++) {
+        const a = this[j];
+        const b = this[j + 1];
+
+        if (!compareFunction) {
+          if (String(a) > String(b)) {
+            const temp = a;
+
+            this[j] = b;
+            this[j + 1] = temp;
+          }
+        } else {
+          if (compareFunction(a, b) > 0) {
+            const temp = a;
+
+            this[j] = b;
+            this[j + 1] = temp;
+          }
+        }
+      }
+    }
```

Comment:
The nested loops implement a bubble sort algorithm, which is acceptable for this task. However, note that bubble sort is not the most efficient sorting algorithm for large datasets. Consider using more efficient algorithms like quicksort or mergesort for better performance if needed.